### PR TITLE
Thorium

### PR DESCRIPTION
**Pull request by thatch45**
This is the beginning of a next generation reactor that uses the salt state system for data flow processing on the event bus within an engine. As it is built out it should allow for very complicated reactors/engines that can be executed entirely via sls files

### CODE REVIEW

**Comment by thatch45** [comment]
Ok, today was a good day, this still needs a lot of testing though. But the main bones are in.

File: salt/thorium/__init__.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        to iterate over
+        '''
+        ret = {}
+        err = []
+        try:
+            top = self.get_top()
+        except SaltRenderError as err:
+            return ret
+        except Exception:
+            trb = traceback.format_exc()
```
Comment (s0undt3ch) [inline salt/thorium/__init__.py]: We should also log the exception message and show the traceback at the debug level(`log.error(msg, exc_info_on_loglevel=logging.DEBUG)`)

**Comment by s0undt3ch** [comment]
Extra kudos for taking care of setup.py too!!!

Does that added setup option show in `python setup.py install --help` or `python setup.py --help`?

**Comment by cachedout** [comment]
This will need to be ported into the Boron branch along with lint fixes from #30876 as soon as it is merged.

**Comment by thatch45** [comment]
Yep, we froze JUST before I was able to get this ready :)

**Pull request merged**